scripts/irods/lib.py

```diff
@@ -751,3 +760,12 @@ def make_arbitrary_file(f_name, f_size, buffer_size=32*1024*1024):
             out.write(bytearray(buffer))
 
             bytes_written += to_write
+
+def get_resource_parent(session, resource_name):
+    parent_id = session.run_icommand(['iquest', '%s',
+        "select RESC_PARENT where RESC_NAME = '{}'"
+        .format(resource_name)])[0].strip()
+
+    return session.run_icommand(['iquest', '%s',
+        "select RESC_NAME where RESC_ID = '{}'"
+        .format(parent_id)])[0].strip()
```

Comment:
This could return a tuple containing the resc id and name. If not, then consider adding `_name` to the function name.

Feel free to ignore this.

Reply:
I think I'd rather split it in two, in that case: `get_resource_parent_name` and `get_resource_parent_id`. Hold please.

Reply:
Well, before I make the change, just know that this has already been merged in 4-2-stable as-is. Are we okay with the split?

Reply:
Let's leave it as is.

Consider creating an issue to revisit this in main and 4-3-stable.

Reply:
Yep, makes sense. Created https://github.com/irods/irods/issues/7029.